gitpod-io/gitpod
# [papi] add api to allow organization owner to create a temporary token

## Description


- [x] Need to rebase after https://github.com/gitpod-io/gitpod/pull/19337 is merged

## Related Issue(s)

Fixes EXP-1084, EXP-1085

## How to test

**Without OIDC, test script mentioned below should failed. Since there's no organization owned users even you have Feature Flag enabled.**


- Try test with test script https://github.com/gitpod-io/gitpod/blob/9c645b1b3d7a1a6fde622d4aadae5f34c6a35b01/test/tests/smoke-test/papi_create_temp_token_test.go#L22-L33

<details>
<summary>Create an OIDC preview env youself</summary>

- Open this PR with Gitpod, create another branch based on current one
- Exec command
```
export GITPOD_WITH_DEDICATED_EMU=true
leeway run dev:preview
```
- Follow [internal Notion page](https://www.notion.so/gitpod/Test-SSO-for-Organizations-c3cb57b4ff1c40da92e268bf3f590b6c) to get SSO info
- Exec command
```
previewctl admin credentials create
kubectl rollout restart deploy/server

# access link print via previewctl
```
- Setup our SSO
</details>

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
